qualtran/bloqs/basic_gates/hadamard.py

```diff
@@ -0,0 +1,71 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property, reduce
+from typing import Dict, Tuple
+
+import cirq
+import numpy as np
+import quimb.tensor as qtn
+from attrs import frozen
+
+from qualtran import Bloq, Signature, SoquetT
+from qualtran.cirq_interop import CirqQuregT
+
+_HADAMARD = np.array([[1.0, 1.0], [1.0, -1.0]]) / (2**0.5)
+
+
+@frozen
+class Hadamard(Bloq):
+    """The Hadamard gate.
+
+    Args:
+        bitsize: The number qubits to add a Hadamard gate to.
+
+    Registers:
+     - q: bitsize-bit register.
+    """
+
+    bitsize: int = 1
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build(q=self.bitsize)
+
+    def as_cirq_op(
+        self, qubit_manager: 'cirq.QubitManager', q: 'CirqQuregT'
+    ) -> Tuple['cirq.Operation', Dict[str, 'CirqQuregT']]:
+        import cirq
+
+        if len(q) == 1:
+            (q,) = q
+            return cirq.H(q), {'q': [q]}
+        else:
+            return cirq.H.on_each(q), {'q': q}
```

Comment:
This would return a list of operations instead of one operation. Currently in qualtran, I've been a stickler for this returning exactly one operation so we don't start writing decompositions that are accessible during cirq conversion but which are not accessible via `decompose_bloq()`. In #344 (for unrelated state fixes) -- in the case where there's literally not a single cirq operation but we still want cirq conversion I use `CircuitOperation` so the structure isn't (automatically) lost. The cirq user can flatten it or whatever after the fact